x-pack/test/security_solution_cypress/cypress/e2e/investigations/alerts/building_block_alerts.cy.ts

```diff
@@ -9,36 +9,34 @@ import { getBuildingBlockRule } from '../../../objects/rule';
 import { OVERVIEW_ALERTS_HISTOGRAM_EMPTY } from '../../../screens/overview';
 import { HIGHLIGHTED_ROWS_IN_TABLE } from '../../../screens/rule_details';
 import { OVERVIEW } from '../../../screens/security_header';
-import { goToRuleDetails } from '../../../tasks/alerts_detection_rules';
 import { createRule } from '../../../tasks/api_calls/rules';
 import { cleanKibana } from '../../../tasks/common';
 import { waitForAlertsToPopulate } from '../../../tasks/create_new_rule';
 import { login, visitWithoutDateRange } from '../../../tasks/login';
 import { waitForTheRuleToBeExecuted } from '../../../tasks/rule_details';
 import { navigateFromHeaderTo } from '../../../tasks/security_header';
-import { DETECTIONS_RULE_MANAGEMENT_URL } from '../../../urls/navigation';
+import { ruleDetailsUrl } from '../../../urls/navigation';
 
 const EXPECTED_NUMBER_OF_ALERTS = 5;
 
 describe(
   'Alerts generated by building block rules',
   { tags: ['@ess', '@brokenInServerless'] },
   () => {
-    before(() => {
+    beforeEach(() => {
       cy.task('esArchiverLoad', { archiveName: 'auditbeat_big' });
       cleanKibana();
       login();
+      createRule(getBuildingBlockRule()).then((rule) =>
+        visitWithoutDateRange(ruleDetailsUrl(rule.body.id))
+      );
     });
-    beforeEach(() => {
-      createRule(getBuildingBlockRule());
-    });
-    after(() => {
+
+    afterEach(() => {
```

Comment:
Is there a reason for using `beforeEach` and `afterEach` here?
Especially the unload task should not be executed as often, iirc.

Reply:
In this particular test suite is doesn't make any difference as there is just one test. Though generally speaking it rather depends on the tests if `before` is enough or some changed data has to be wiped out before each test. When more tests are added it should be taken into account. It looks like `esArchiverLoad` can safely happen once per test suite so I rolled back some changes but left `cleanKibana()` and rule creation in `beforeEach`.